go/oasis-node/cmd/common/metrics/config/config.go

```diff
@@ -0,0 +1,35 @@
+// Package config implements global configuration options.
+package config
+
+import "time"
+
+// Config is the metrics configuration structure.
+type Config struct {
+	// Metrics mode (none, pull).
+	Mode string `yaml:"mode"`
+	// Metrics pull address.
+	Address string `yaml:"address"`
+
+	// Metrics push job name (debug-only).
+	JobName string `yaml:"job_name,omitempty"`
+	// Metrics push instance labels (debug-only).
+	Labels map[string]string `yaml:"labels,omitempty"`
+	// Metrics push interval (debug-only).
+	Interval time.Duration `yaml:"interval,omitempty"`
+}
+
+// Validate validates the configuration settings.
+func (c *Config) Validate() error {
+	return nil
```

Comment:
I've added some validation here, but I don't want to make it too strict because then if someone has all the fields set and wants to temporarily disable metrics, the validation shouldn't complain about the (now) extra fields...